long-running-tasks/src/test/java/io/streamnative/pulsar/recipes/task/Soak.java

```diff
@@ -0,0 +1,398 @@
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.shade.com.fasterxml.jackson.annotation.JsonIgnore;
+import org.apache.pulsar.shade.com.google.common.collect.Streams;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.testcontainers.containers.PulsarContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * This is a soak test that is intended to be executed indpendently and with user-defined parameters
+ * to simulate their expected workloads.
+ */
+@Disabled
+@Slf4j
+@Testcontainers
+public class Soak {
+  private static final DockerImageName pulsarImage =
+      DockerImageName.parse("apachepulsar/pulsar").withTag("2.10.0");
+
+  @Container
+  private static final PulsarContainer pulsar =
+      new PulsarContainer(pulsarImage) {
+        @Override
+        protected void configure() {
+          super.configure();
+          withStartupTimeout(Duration.ofMinutes(3));
+          withEnv("PULSAR_PREFIX_delayedDeliveryTickTimeMillis", "5");
+        }
+      };
+
+  private final Clock clock = Clock.systemUTC();
+  private final Schema<Task> taskSchema = Schema.JSON(Task.class);
+  private final Schema<TaskMetadata> metadataSchema = Schema.JSON(TaskMetadata.class);
+  private final Schema<Result> resultSchema = Schema.JSON(Result.class);
+
+  private PulsarClient client;
+  private Producer<Task> producer;
+  private Consumer<TaskMetadata> consumer;
+
+  private void createResources(String taskTopic) throws Exception {
+    client = PulsarClient.builder().serviceUrl(pulsar.getPulsarBrokerUrl()).build();
+    producer = client.newProducer(taskSchema).topic(taskTopic).enableBatching(false).create();
+    consumer =
+        client
+            .newConsumer(metadataSchema)
+            .topic(taskTopic + "-state")
+            .subscriptionName(randomUUID().toString())
+            .subscribe();
+  }
+
+  @AfterEach
+  void afterEach() throws Exception {
+    producer.close();
+    consumer.close();
+  }
+
+  @Test
+  @Timeout(value = 4, unit = HOURS)
+  void soak() throws Exception {
+    // Modify these parameters to match your expected workload.
+    var soakTimeout = Duration.ofHours(4);
+    var workerCount = 15;
+    var keepAliveInterval = Duration.ofSeconds(1);
+    var taskCount = 100;
+    var meanTaskTime = Duration.ofMinutes(5);
+    var maxRetries = 3;
+    var taskDurationDistribution =
+        new NormalDistribution(meanTaskTime.toSeconds(), meanTaskTime.toSeconds() / 2.0);
+    var taskOutcomeDistribution =
+        new EnumeratedDistribution<>(
+            List.of(
+                Pair.create(RESULT, 0.9),
+                Pair.create(EXCEPTION, 0.75),
+                Pair.create(TIMEOUT, 0.25)));
+
+    // Begin test
+    var taskTopic = randomUUID().toString();
```

Comment:
Perhaps add a prefix to make it distinguishable. Is it the intention to use this to test Pulsar?